### DISCUSSION TITLE
Slint 1.5: Embracing Android, Improving Live-Preview, and introducing Pythonic Slint

### DISCUSSION THREAD

**Discussion by ogoffart** [slint-ui/slint#4845] [Blog Comments]
Release announcement with the highlights: https://slint.dev/blog/slint-1.5-released

Detailed list of changes: [ChangeLog](https://github.com/slint-ui/slint/blob/master/CHANGELOG.md#150---2024-03-14)

<hr /><em>This discussion was created from the release <a href='https://github.com/slint-ui/slint/releases/tag/v1.5.0'>1.5.0</a>.</em>

**Comment by Jhonfunk** [comment]
```cmd
cargo update
    Updating index
error: failed to select a version for the requirement slint = "^1.5.0"
candidate versions found which didn't match: 1.4.1, 1.4.0, 1.3.2, ...
location searched:  index (which is replacing registry crates-io)
required by package perhaps a crate was updated and forgotten to be re-vendored?
```

**Reply by tronical** [reply]
Hm that’s odd. Could you post your entire `Cargo.toml` here as well as the complete error message? Sometimes cargo gives a deeper explanation. Thanks:)

**Reply by Jhonfunk** [reply]
Everything is fine now, it seems that the mirrors just weren't updated at the time.

**Reply by tronical** [reply]
Phew :)

**Comment by ghost** [comment]
Is there an apk file from the blog post somewhere?

**Reply by ogoffart** [reply]
Our nightly build builds the apk.
This is the link to the one from the 1.5.1 release: https://github.com/slint-ui/slint/actions/runs/8355777596
If you scroll down to "Artifacts", you can download android-demo.zip that contains the apk for the energy-monitor (the one from the blog post) and the todo example.
(you need to be logged in github)

**Reply by J05HI** [reply]
Hello @ogoffart 
first of all thank you for the amazing project!
I have one Problem on Android. The scrolling doesn't behave natively. It stops immediately after lifting the finger from the screen. I'm on a Pixel 7 Pro.

**Comment by DennisZhangOiler** [comment]
Can I use python to build a slint app and distribute it to Android?

**Reply by ogoffart** [reply]
At the moment, this might not work out of the box.
The missing part is mainly the build system part to bundle everything in a apk, including the slint-backend-android-activity
